test/tracer.spec.ts

```diff
@@ -489,6 +492,257 @@ describe('Autoblocks Tracer', () => {
+            threshold: { gt: 0, lte: 1 },
+            metadata: {
+              'some-metadata-key': 'some-metadata-value',
+            },
+          });
+        }
+      }
+      const tracer = new AutoblocksTracer('mock-ingestion-key');
+      const { traceId } = await tracer.sendEvent('mock-message', {
+        properties: {
+          'my-prop-key': 'my-prop-value',
+        },
+        traceId: 'my-trace-id',
+        evaluators: [new MyEvaluator()],
+      });
+
+      expect(traceId).toEqual('mock-trace-id');
+
+      expectPostRequest({
+        message: 'mock-message',
+        traceId: 'my-trace-id',
+        timestamp,
+        properties: {
+          'my-prop-key': 'my-prop-value',
+          evaluations: [
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: {
+                gt: 0,
+                lte: 1,
+              },
+              metadata: {
+                'some-metadata-key': 'some-metadata-value',
+              },
+              evaluatorExternalId: mockEvaluatorExternalId,
+            },
+          ],
+        },
+      });
+    });
+
+    it('handles async evaluators', async () => {
+      class MyEvaluator extends BaseEventEvaluator {
+        id = mockEvaluatorExternalId;
+
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        async evaluateEvent(args: { event: TracerEvent }): Promise<Evaluation> {
+          return Promise.resolve({ score: 0.9 });
+        }
+      }
+      const tracer = new AutoblocksTracer('mock-ingestion-key');
+      const { traceId } = await tracer.sendEvent('mock-message', {
+        evaluators: [new MyEvaluator(), new MyEvaluator()],
+      });
+
+      expect(traceId).toEqual('mock-trace-id');
+
+      expectPostRequest({
+        message: 'mock-message',
+        traceId: undefined,
+        timestamp,
+        properties: {
+          evaluations: [
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: undefined,
+              metadata: undefined,
+              evaluatorExternalId: mockEvaluatorExternalId,
+            },
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: undefined,
+              metadata: undefined,
+              evaluatorExternalId: mockEvaluatorExternalId,
+            },
+          ],
+        },
+      });
+    });
+
+    it('handles multiple evaluators', async () => {
+      class MyEvaluator extends BaseEventEvaluator {
+        id = mockEvaluatorExternalId;
+
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        evaluateEvent(args: { event: TracerEvent }): Evaluation {
+          return {
+            score: 0.9,
+          };
+        }
+      }
+      const tracer = new AutoblocksTracer('mock-ingestion-key');
+      const { traceId } = await tracer.sendEvent('mock-message', {
+        evaluators: [new MyEvaluator(), new MyEvaluator(), new MyEvaluator()],
```

Comment:
i think this test would be more effective if each evaluator had its own ID. as-is we're not really testing that all 3 evaluators were ran b/c they have the same exact data